### PR TITLE
Update cross validation function

### PR DESCRIPTION
**Pull request by JieXiong9119**
Fix testing data generation
Add Monte Carlo cross-validation function

### CODE REVIEW

File: run/test_run.py [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-    weather = 'Norfolk',
-    labeling_methodology = 'Simple',
-    feature_selection_methodology = 'Embedded',
-    aggregate_n_runs = 6 * 24,
-    number_of_trees = 4,
-    randomseed=20210813
-    )
-
-test_run.whole_process_training_and_testing()
+randomruns = 10
```
Comment (JieXiong9119) [inline run/test_run.py:4]: Change this number to define the number of runs.

**Comment by JanghyunJK** [comment]
not done with the entire review, but is it possible to keep all jupyter notebook files under FDD_RF_Repo\JupyterNotebook folder? I'm mostly referring to this file `run/results/Results.ipynb`. and preferring to keep all jupyter notebooks in one place if these are mostly used for testing (before being reflected in the actual workflow).

**Comment by JieXiong9119** [comment]
> not done with the entire review, but is it possible to keep all jupyter notebook files under FDD_RF_Repo\JupyterNotebook folder? I'm mostly referring to this file `run/results/Results.ipynb`. and preferring to keep all jupyter notebooks in one place if these are mostly used for testing (before being reflected in the actual workflow).

I'm good with that. The only jupyter notebook I used is the same file `run/results/Results.ipynb` - I just added a portion for visualizing the cross-validation results in boxplot.

**Comment by JanghyunJK** [comment]
k, I went through all,
- tested on my side
- saw testing data automatically being generated with 20% split
- tried 10 trials with 400 trees and all went well
- also updated figure in slide 102
- suggested some structural changes in this round
- one thing I want to fix later is to change this two same folder of Norfolk into just one folder but anyways
- I think all changes look good, so see if any of my changes are not working for you
- and if all works fine, just merge this branch to `estcp_cross_validation` branch
- and then submit another PR to merge `estcp_cross_validation` to `main`, so that we could discuss this with Liang.

File: FDD_RF_Module/FDD_RF_Modeling.py [hunk 2]
```diff
@@ -67,6 +67,19 @@ def CDDR_tot(self, Real_label, Pred_label):
         CDDR_tot = CD / CP
         return CDDR_tot
 
+    def TPR_FPR_tot(self, Real_label, Pred_label):
```
Comment (JieXiong9119) [inline FDD_RF_Module/FDD_RF_Modeling.py:70]: @JanghyunJK I added this part to enable detection performance evaluation, after we have fault-free data included. For now the output for TPR & FPR will be all 1 & 0.

**Review by JieXiong9119 — COMMENTED** [review]
> k, I went through all,
> 
> * tested on my side
> * saw testing data automatically being generated with 20% split
> * tried 10 trials with 400 trees and all went well
> * also updated figure in slide 102
> * suggested some structural changes in this round
> * one thing I want to fix later is to change this two same folder of Norfolk into just one folder but anyways
> * I think all changes look good, so see if any of my changes are not working for you
> * and if all works fine, just merge this branch to `estcp_cross_validation` branch
> * and then submit another PR to merge `estcp_cross_validation` to `main`, so that we could discuss this with Liang.

It works good to me. I just commit a small change to enable detection performance evaluation, after we have fault-free data included. For now the output for TPR & FPR will be all 1 & 0.
If you think it's good, we can merge it!

**Comment by JanghyunJK** [comment]
awesome! so TPR/FPR are going into the log results now. got it. worked on my computer, so merging!

**Pull request merged**